implementations/rust/ockam/ockam_identity/src/secure_channels/secure_channels.rs

```diff
@@ -108,10 +112,27 @@ impl SecureChannels {
             options.trust_context,
             Some(route),
             Some(options.timeout),
+            Arc::clone(&flag),
             Role::Initiator,
         )
         .await?;
 
+        let self_clone = self.clone();
+        let ctx_clone = ctx.async_try_clone().await?;
+        let addr = addresses.encryptor.clone();
+        tokio::spawn(async move {
+            let flag_value = Arc::clone(&flag);
+            while !flag_value.load(Ordering::Relaxed) {
```

Comment:
You are already checking below:
```
if !flag_value.load(Ordering::Relaxed) {
    self_clone.stop_secure_channel(&ctx_clone, &addr).await.unwrap();
    break;
}
```
If the flag changes state just after you set it to false (highly unlikely by possible), the loop would exit in the main condition and `self_clone.stop_secure_channel(&ctx_clone, &addr).await.unwrap();` would not be called.